src/modules/html/producer/html_producer.cpp

```diff
@@ -524,7 +536,7 @@ class html_producer : public core::frame_producer
 
     std::wstring print() const override { return L"html[" + url_ + L"]"; }
 
-    core::monitor::state state() const override { return state_; }
+    core::monitor::state state() const override { return client_->state(); }
```

Comment:
You should check client_ for null here, like all the other methods do.
I dont know how it could ever be null, but consistency